### PR TITLE
[SPIKE] API v1.1 defer confirm deferral of applications

### PR DESCRIPTION
**Pull request by despo**
## Link to Trello card
https://trello.com/c/6JWqXu6i/4663-%F0%9F%8F%88-api-v11-defer-confirm-deferral-of-applications

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/presenters/concerns/application_transitioning_module.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# to remove once Notes and Withdrawals add the required fields
+module ApplicationTransitioningModule
```
Comment (cwrw) [inline app/presenters/concerns/application_transitioning_module.rb:2]: I don't think we should be introducing this at this stage, I think we can wait till the API version is released, then we can start validating against the schema, something to discuss today in the API meeting

File: app/services/reinstate_deferred_offer.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class ReinstateDeferredOffer
```
Comment (cwrw) [inline app/services/reinstate_deferred_offer.rb:1]: can we use the wording confirm deferred offer instead, as we are moving away from reinstate?

**Review by cwrw — COMMENTED** [review]
can we break up this pr into defer and confirm please, so it's easier to review each one in isolation?

**Comment by despo** [comment]
> can we break up this pr into defer and confirm please, so it's easier to review each one in isolation?

@cwrw I'm in the process of doing that. Already opened two PRs and working through refactoring the existing services before implementing Defer and ConfirmDeferred offer.

**Pull request closed without merging**